lib/stitch_fix/log_weasel/transaction.rb

```diff
@@ -19,6 +23,13 @@ def self.id=(id)
       def self.id
         Thread.current[:log_weasel_id]
       end
+
+      def self.scope
```

Comment:
This will work, but I think you could avoid the regex match if you added a getter for `key`, which gets appended to the log weasel ID in the `self.create` method above. This also clears up the naming problem. I'm thinking something along the lines of:
```
def self.create(key = nil)
  Thread.current[:log_weasel_id] = "#{ULID.generate}#{self.append_key(key)}"
end

def self.append_key(key)
  Thread.current[:log_weasel_key] = (key ? "-#{key}" : "")
end

def self.key
  Thread.current[:log_weasel_key] 
end
```

Thoughts?

Reply:
@brettfishman if I'm reading the logic in `request_context` right, it'll translate our local thread variables into headers, meaning that once the `log_weasel_key` is generated it would be propagated as a header? 

If so, that works, but I think the only case where it falls apart is in a request coming from either iOS or Android Native. We could still potentially use a regex to check for that when `log_weasel_key` is not set; that would take it down to a much smaller set of operations where we'd need a regex.

Reply:
EDIT: Maybe `self.create_key` rather than `self.append_key` would be better.

Reply:
Ah, you made me think about our `stitch_fix/concurrency` gem and how we'll need to propagate the LW key, [like we do the LWID](https://github.com/stitchfix/concurrency/blob/7fe6be279c3b9c06f18ebd5c29fcb3a2ddca918e/lib/stitch_fix/concurrency/safe_future.rb#L83), in `safe_future.rb` - ugh, even more overhead with this solution.

Your solution avoids all of that, and avoids adding additional global state. 🤔 

Ok, I'm coming around to this approach. It might come down to naming for me. How do you feel about `self.key` rather than `self.scope`? My rationale is, the `key` is a log weasel concept, and `scope` is more of a Stitch Fix invented thing. I think I find it less surprising to have a method that parses the LW key from a LW ID, and exposes it via a `self.key` method, rather than a `self.scope method.

Reply:
I'm preferable to `key` as well; neither `scope` nor `source_app` felt quite right. I'll get that updated.
Also I think I need to learn more about `stitch_fix/concurrency` and how a lot of these core gems work in combination with each other.

Reply:
`stitch_fix/concurrency` enables an app to make concurrent requests to another app, which is useful in a microservices architecture when a given app needs to talk to N number of services to fulfill a given request (e.g. GraphQL). One thing we ran into is, if there was any global state in the parent request thread, it wouldn't get passed along to child request threads. So we needed to enhance our `stitch_fix/concurrency` library to support specific use cases (e.g. like propagating the LWID) where state was managed globally.